java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java

```diff
@@ -4193,6 +4193,206 @@ void testContainsReEmptyInput() {
     }
   }
 
+  @Test
+  void testLike() {
+    Scalar defaultEscape = Scalar.fromString("\\");
+    // Basic patterns (with null string entry)
+    try (ColumnVector testStrings = ColumnVector.fromStrings("", null, "ovér the", "lazy @dog", "1234", "00:0:00");
+         Scalar patternString1 = Scalar.fromString("");
+         Scalar patternString2 = Scalar.fromString("1234");
+         ColumnVector res1 = testStrings.like(patternString1, defaultEscape);
+         ColumnVector res2 = testStrings.like(patternString2, defaultEscape);
+         ColumnVector expected1 = ColumnVector.fromBoxedBooleans(
+           true, null, false, false, false, false);
+         ColumnVector expected2 = ColumnVector.fromBoxedBooleans(
+           false, null, false, false, true, false)) {
+      assertColumnsAreEqual(expected1, res1);
+      assertColumnsAreEqual(expected2, res2);
```

Comment:
If you have multiple separate tests, let's have separate inner `try()` blocks for each of them.